src/app/clusters/zone-management-server/zone-management-server.h

```diff
@@ -0,0 +1,356 @@
+     *                                          This value is specified by the device manufacturer.
+     * @param aTwoDCartesianMax                 The maximum X and Y points that are allowed for TwoD Cartesian Zones.
+     *
+     */
+    ZoneMgmtServer(ZoneMgmtDelegate & aDelegate, EndpointId aEndpointId, const BitFlags<Feature> aFeatures,
+                   const BitFlags<OptionalAttribute> aOptionalAttrs, uint8_t aMaxUserDefinedZones, uint8_t aMaxZones,
+                   uint8_t aSensitivityMax, const TwoDCartesianVertexStruct & aTwoDCartesianMax);
+
+    ~ZoneMgmtServer() override;
+
+    /**
+     * @brief Initialise the Zone Management server instance.
+     * This function must be called after defining a ZoneMgmtServer class object.
+     * @return Returns an error if the given endpoint and cluster ID have not been enabled in zap or if the
+     * CommandHandler or AttributeHandler registration fails, else returns CHIP_NO_ERROR.
+     * This method also checks if the feature setting is valid, if invalid it will return CHIP_ERROR_INVALID_ARGUMENT.
+     */
+    CHIP_ERROR Init();
+
+    bool HasFeature(Feature feature) const;
+
+    bool SupportsOptAttr(OptionalAttribute aOptionalAttr) const;
+
+    // Attribute Setters
+    CHIP_ERROR SetSensitivity(uint8_t aSensitivity);
+
+    // Attribute Getters
+    const std::vector<ZoneInformationStorage> & GetZones() const { return mZones; }
+
+    const std::vector<ZoneTriggerControlStruct> & GetTriggers() const { return mTriggers; }
+
+    uint8_t GetMaxUserDefinedZones() const { return mMaxUserDefinedZones; }
+    uint8_t GetMaxZones() const { return mMaxZones; }
+    uint8_t GetSensitivityMax() const { return mSensitivityMax; }
+    uint8_t GetSensitivity() const { return mSensitivity; }
+    const TwoDCartesianVertexStruct & GetTwoDCartesianMax() const { return mTwoDCartesianMax; }
+
+    CHIP_ERROR AddZone(const ZoneInformationStorage & zone);
+    CHIP_ERROR UpdateZone(uint16_t zoneId, const ZoneInformationStorage & zone);
+    CHIP_ERROR RemoveZone(uint16_t zoneId);
+
+    Protocols::InteractionModel::Status AddOrUpdateTrigger(const ZoneTriggerControlStruct & trigger);
+    Protocols::InteractionModel::Status RemoveTrigger(uint16_t zoneId);
+
+    // Send Zone events
+    Protocols::InteractionModel::Status GenerateZoneTriggeredEvent(uint16_t zoneID, ZoneEventTriggeredReasonEnum triggerReason);
+    Protocols::InteractionModel::Status GenerateZoneStoppedEvent(uint16_t zoneID, ZoneEventStoppedReasonEnum stopReason);
+
+private:
+    ZoneMgmtDelegate & mDelegate;
+    EndpointId mEndpointId;
+    const BitFlags<Feature> mFeatures;
+    const BitFlags<OptionalAttribute> mOptionalAttrs;
+
+    // Attributes
+    const uint8_t mMaxUserDefinedZones;
+    const uint8_t mMaxZones;
+    const uint8_t mSensitivityMax;
+    const TwoDCartesianVertexStruct mTwoDCartesianMax;
+    uint8_t mUserDefinedZonesCount = 0;
+
+    std::vector<ZoneInformationStorage> mZones;
```

Comment:
Is it correct that `mZones` is stored in-memory & never loaded?

I.e. whose responsibility is it to persist zones?

Reply:
This probably was a bad oversight. Both Zones and Triggers ought to be persisted I think. Will update here and post a Spec PR.